core/jobs.py

```diff
@@ -1411,11 +1405,6 @@ def _clear_inactive_realtime_layer(cls, latest_created_on_datetime):
     @classmethod
     def _kickoff_batch_job(cls):
         """Create and enqueue a new batch job."""
-        if job_models.JobModel.do_unfinished_jobs_exist(cls.__name__):
```

Comment:
I'm not sure why the previous comment on this line was marked as resolved -- it doesn't seem resolved to me.

That said, I think these lines can be deleted based on the reasoning provided. However, I am concerned about "First I did start_computation and again did start_computation. No exception was raised." This seems like something that should be tested and fixed?

Reply:
> "First I did start_computation and again did start_computation. No exception was raised."

This happens when I remove the lines below in https://github.com/oppia/oppia/blob/d1d8e53bed78e7716b8af30752e7ff278bec5aa4/core/jobs.py#L1476

Reply:
I also added a test for this: `test_cannot_start_new_job_while_existing_job_still_running`

Reply:
I'm sorry, I don't understand. Can you elaborate? I don't know which lines you're removing, and I don't understand the cause of the error.

Reply:
I am removing the lines 1414-1418 as we will get an error from [this line](https://github.com/oppia/oppia/blob/d1d8e53bed78e7716b8af30752e7ff278bec5aa4/core/jobs.py#L1476) if we are to start a new job with another one running. I stated that if I remove this [line](https://github.com/oppia/oppia/blob/d1d8e53bed78e7716b8af30752e7ff278bec5aa4/core/jobs.py#L1476), I am not getting an exception -- means I am doing start_computation and again doing start_computation but still not getting an error.

Reply:
Ah ok. So, you are not planning to remove line 1476, right?

Reply:
No I am not. I have also added a test for that line(1476)

Reply:
OK! Sounds good. Thanks for clarifying.